server/plugin.go

```diff
@@ -441,26 +441,42 @@ func (p *Plugin) syncUsers() {
 
 		p.API.LogDebug("Running sync user job for user with email", "email", msUser.Mail)
 
-		mmUser, ok := mmUsersMap[msUser.Mail]
+		mmUser, isUserPresent := mmUsersMap[msUser.Mail]
 
-		if !msUser.IsAccountEnabled {
-			if ok && mmUser.RemoteId != nil && *mmUser.RemoteId != "" && strings.HasPrefix(mmUser.Username, "msteams_") {
-				p.API.LogDebug("Deactivating the Mattermost user account", "Email", msUser.Mail)
-				if err := p.API.UpdateUserActive(mmUser.Id, false); err != nil {
-					p.API.LogError("Unable to deactivate the Mattermost user account", "Email", mmUser.Email, "Error", err.Error())
+		if isUserPresent && isSyncUser(mmUser) {
+			if !msUser.IsAccountEnabled {
+				// Deactivate the active Mattermost user corresponding to MS Teams user.
+				if mmUser.DeleteAt == 0 {
+					p.API.LogDebug("Deactivating the Mattermost user account", "Email", msUser.Mail)
+					if err := p.API.UpdateUserActive(mmUser.Id, false); err != nil {
+						p.API.LogError("Unable to deactivate the Mattermost user account", "Email", mmUser.Email, "Error", err.Error())
+					}
+
+					continue
 				}
-			}
+			} else {
+				// Activate the deactive Mattermost user corresponding to MS Teams user.
+				if mmUser.DeleteAt != 0 {
+					p.API.LogDebug("Activating the inactive user", "Email", msUser.Mail)
+					if err := p.API.UpdateUserActive(mmUser.Id, true); err != nil {
+						p.API.LogError("Unable to activate the user", "Email", msUser.Mail, "Error", err.Error())
+					}
 
-			continue
+					continue
+				}
+			}
 		}
 
 		if msUser.Type == "Guest" {
+			// Check if syncing of MS Teams guest user is disabled.
 			if !syncGuestUsers {
-				if !ok {
+				// Skip syncing of MS Teams guest user.
```

Comment:
I think it will be easier to read and understand if you reverse the if condition and move the comment inside `if` statement.